lib/include/openamp/virtio_mmio.h

```diff
@@ -173,6 +173,33 @@ static struct vring __vring_##name = { \
  * @note TODO: look into MMIO_QUEUE_ALIGN
  */
 
+#if defined(HVL_VIRTIO)  /* vrings in individual sections */
+
+#define VQ_DECLARE(name, n, align) \
+	static char __vrbuf_##name[VRING_SIZE(n, align)] __aligned(4096) \
+	__attribute((__section__(".shared.vring."#name))); \
```

Comment:
Look like only __vrbuf_##name is different
could you define something like this instead?
```
#if defined(HVL_VIRTIO)  /* vrings in individual sections */
#define __VRBUF(name, section_name)   \
	 static char __vrbuf_##name[VRING_SIZE(n, align)]  __aligned(4096) \
	__attribute((__section__(".shared.vring."#section_name)));
#else
	 static char __vrbuf_##name[VRING_SIZE(n, align)]  __aligned(4096);
#endif
```
This would help to see delta

Reply:
Moved to new PR.